src/main/java/baritone/cache/FasterWorldScanner.java

```diff
@@ -0,0 +1,249 @@
+    }
+
+    @Override
+    public List<BlockPos> scanChunk(IPlayerContext ctx, BlockOptionalMetaLookup filter, ChunkPos pos, int max, int yLevelThreshold) {
+        return scanChunkInternal(ctx, filter, pos).limit(max).collect(Collectors.toList());
+    }
+
+    @Override
+    public int repack(IPlayerContext ctx) {
+        return this.repack(ctx, 40);
+    }
+
+    @Override
+    public int repack(IPlayerContext ctx, int range) {
+        IChunkProvider chunkProvider = ctx.world().getChunkProvider();
+        ICachedWorld cachedWorld = ctx.worldData().getCachedWorld();
+
+        BetterBlockPos playerPos = ctx.playerFeet();
+
+        int playerChunkX = playerPos.getX() >> 4;
+        int playerChunkZ = playerPos.getZ() >> 4;
+
+        int minX = playerChunkX - range;
+        int minZ = playerChunkZ - range;
+        int maxX = playerChunkX + range;
+        int maxZ = playerChunkZ + range;
+
+        int queued = 0;
+        for (int x = minX; x <= maxX; x++) {
+            for (int z = minZ; z <= maxZ; z++) {
+                Chunk chunk = chunkProvider.getLoadedChunk(x, z);
+
+                if (chunk != null && !chunk.isEmpty()) {
+                    queued++;
+                    cachedWorld.queueForPacking(chunk);
+                }
+            }
+        }
+
+        return queued;
+    }
+
+    // ordered in a way that the closest blocks are generally first
+    public static List<ChunkPos> getChunkRange(int centerX, int centerZ, int chunkRadius) {
+        List<ChunkPos> chunks = new ArrayList<>();
+        // spiral out
+        chunks.add(new ChunkPos(centerX, centerZ));
+        for (int i = 1; i < chunkRadius; i++) {
+            for (int j = 0; j <= i; j++) {
+                chunks.add(new ChunkPos(centerX - j, centerZ - i));
+                if (j != 0) {
+                    chunks.add(new ChunkPos(centerX + j, centerZ - i));
+                    chunks.add(new ChunkPos(centerX - j, centerZ + i));
+                }
+                chunks.add(new ChunkPos(centerX + j, centerZ + i));
+                if (j != 0 && j != i) {
+                    chunks.add(new ChunkPos(centerX - i, centerZ - j));
+                    chunks.add(new ChunkPos(centerX + i, centerZ - j));
+                    chunks.add(new ChunkPos(centerX - i, centerZ + j));
+                    chunks.add(new ChunkPos(centerX + i, centerZ + j));
+                }
+            }
+        }
+        return chunks;
+    }
+
+    private List<BlockPos> scanChunksInternal(IPlayerContext ctx, BlockOptionalMetaLookup lookup, List<ChunkPos> chunkPositions, int maxBlocks) {
+        assert ctx.world() != null;
+        try {
+            Stream<BlockPos> posStream = chunkPositions.parallelStream().flatMap(p -> scanChunkInternal(ctx, lookup, p));
+            if (maxBlocks >= 0) {
+                // WARNING: this can be expensive if maxBlocks is large...
+                // see limit's javadoc
+                posStream = posStream.limit(maxBlocks);
```

Comment:
I'm not sure, honestly the speed of the new scanner's fast enough we can probably just deprecate limit in the worldscanner.

Reply:
No, `max` is still needed and if we remove it here users will have to limit themselves after scanning, wasting time on scanning the whole range.
Even if the scanner is fast enough to return all nearby stone fast enough the goal renderer and probably the pathfinder as well won't be happy with the result. (that "all nearby stone" query froze the game when I tried using goal rendering to visualize a cache)

Reply:
there's a secondary cut off to the max in mineprocess already zac

Reply:
Yep, that keeps rendering and pathing fast, but if `WorldScanner` itself causes the slowdown it won't help.
Here's what happens without limiting in `FasterWorldScanner` (you can see it really froze because the cursor stops blinking):

https://user-images.githubusercontent.com/68165024/209749520-381399e9-7a6c-47c6-9b4a-f4fadb91c389.mp4


Actually even with limiting mining stone is laggy as hell with the new scanner while with the old scanner I didn't notice any jitter.

Reply:
I remember having issues with stone on the old scanner so idk... I guess I can look into it...

Reply:
the bulk of the lag isn't my fault for that one... 
![image](https://user-images.githubusercontent.com/6234704/209766650-b4ef4de2-a20a-4b76-84a4-3a2ca4d78e44.png)
this is still with the limit in place